src/EditorFeatures/CSharp/CodeCleanup/CSharpCodeCleanupService.cs

```diff
@@ -0,0 +1,181 @@
+                                          IDEDiagnosticIds.UseExpressionBodyForIndexersDiagnosticId,
+                                          IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId)
+                },
+                {
+                    FeatureOnOffOptions.FixInlineVariableDeclarations,
+                    ImmutableArray.Create(IDEDiagnosticIds.InlineDeclarationDiagnosticId)
+                },
+                {
+                    FeatureOnOffOptions.RemoveUnusedVariables,
+                    ImmutableArray.Create(IDEDiagnosticIds.DeclaredVariableNeverUsedDiagnosticId,
+                                          IDEDiagnosticIds.AssignedVariableNeverUsedDiagnosticId)
+                },
+                {
+                    FeatureOnOffOptions.FixObjectCollectionInitialization,
+                    ImmutableArray.Create(IDEDiagnosticIds.UseObjectInitializerDiagnosticId,
+                                          IDEDiagnosticIds.UseCollectionInitializerDiagnosticId)
+                }
+            };
+
+            return dictionary.ToImmutableDictionary();
+        }
+
+        public async Task<Document> CleanupDocument(Document document, CancellationToken cancellationToken)
+        {
+            var docOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+
+            document = await ApplyCodeFixes(document, docOptions, cancellationToken).ConfigureAwait(false);
+            // do the remove usings after code fix, as code fix might remove some code which can results in unused usings.
+            document = await RemoveSortUsings(document, docOptions, cancellationToken).ConfigureAwait(false);
+
+            return document;
+        }
+
+        private async Task<Document> RemoveSortUsings(Document document, DocumentOptionSet docOptions, CancellationToken cancellationToken)
+        {
+            // remove usings
+            if (docOptions.GetOption(FeatureOnOffOptions.RemoveUnusedUsings))
+            {
+                var removeUsingsService = document.GetLanguageService<IRemoveUnnecessaryImportsService>();
+                if (removeUsingsService != null)
+                {
+                    document = await removeUsingsService.RemoveUnnecessaryImportsAsync(document, cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            // sort usings
+            if (docOptions.GetOption(FeatureOnOffOptions.SortUsings))
+            {
+                document = await OrganizeImportsService.OrganizeImportsAsync(document, cancellationToken).ConfigureAwait(false);
+            }
+
+            return document;
+        }
+
+        private async Task<Document> ApplyCodeFixes(Document document, DocumentOptionSet docOptions, CancellationToken cancellationToken)
+        {
+            var fixAllService = document.Project.Solution.Workspace.Services.GetService<IFixAllGetFixesService>();
+
+            var dummy = new ProgressTracker();
```

Comment:
it feels like we should have an actual progress tracker.  Because this may take a *long* time to run all these fixes on a document, and hte user will just have a blocked experience without knowing what's going on.

Reply:
@CyrusNajmabadi Could you be more specific on which process tracker that you suggest? This ProcessTracker already has a dialog pops up with Cancel button...

Reply:
Is there a progress bar that would track the number of fixes completed?

Reply:
It would increase after every iteration of this loop.

Reply:
nit: consider using a better than "dummy"